### ISSUE TITLE
Failing during initialization

### ISSUE THREAD

**Issue by ppamidi** [ctews/SwiftPulse#1]
Hello,

I'm trying to add pulse animation to a button. And it give me back the below error.

Libraries/LFTPulseAnimation.swift: 12: 7: fatal error: use of unimplemented initializer 'init(layer:)' for class 'xxxxxx.LFTPulseAnimation'

I'm running it in Swift 1.1 and Xcode 6.2.

**Comment by ctews** [comment]
Hi,

could you please provide the code that throws that fatal error :) Then I'll try to reproduce and fix if there's a bug!

Cheers

**Comment by ppamidi** [comment]
Sure,

I'm trying to add the pulsing animation under uibutton centered to its frame. It is placed under the button layer as mentioned in the readme doc.

@IBOutlet weak var beacon_one_button: UIButton!
override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib   
        beacon_one_button.backgroundColor = UIColor(red:0.544,  green:0.427,   blue:0.896, alpha:1)
}

//function called on button click
@IBAction func toggleBeaconOne(sender: AnyObject) {
                beacon_one_button.backgroundColor = UIColor(red:0.909, green:0.411, blue:0.479, alpha:1)
                var halo_layer_one = LFTPulseAnimation(repeatCount: Float.infinity, radius: 300, position: beacon_one_button.center)
                self.view.layer.insertSublayer(halo_layer_one, below: beacon_one_button.layer)
}

Let me know if I'm doing something wrong :)

**Comment by ctews** [comment]
Do you get this at compile time or at run time? I'm currently on Swift v1.2 and it's working fine for me with the same calls. But I used to do my UI programatically. Is the error thrown when you initialize the pulse animation at ```var halo_layer_one = LFTPulseAnimation(...)``` or the line underneath when you try to apply it to the button layer? Did you ```import UIKit```?

Can you make sure that the `weak` declaration of the beacon_one_button isn't raising this problem. A little bit guessing in the wild from my side at the moment, because I can't test it before this or tomorrow evening ;) Just looked at my code that at least compiles without any problems with Swift v1.2 ;)

**Comment by ppamidi** [comment]
Yes, I did imported UIKit and also verified by removing weak reference to button attribute. It still breaks at the initialization step. 

I've included couple of screenshots incase if it helps.

![screen shot 2015-04-23 at 2 49 39 pm](https://cloud.githubusercontent.com/assets/6331675/7304988/b4e2c0fa-e9c8-11e4-870b-607a7001f68d.png)

![screen shot 2015-04-23 at 2 51 35 pm](https://cloud.githubusercontent.com/assets/6331675/7304991/b8f16872-e9c8-11e4-9ea3-70636ba2feaa.png)
